homeassistant/components/sonos/media_player.py

```diff
@@ -517,14 +434,14 @@ async def async_seen(self, player: SoCo) -> None:
             self._seen_timer()
 
         self._seen_timer = self.hass.helpers.event.async_call_later(
-            SEEN_EXPIRE_TIME, self.async_unseen
+            SEEN_EXPIRE_TIME.total_seconds(), self.async_unseen
         )
 
         if was_available:
             return
 
         self._poll_timer = self.hass.helpers.event.async_track_time_interval(
-            self.update, datetime.timedelta(seconds=SCAN_INTERVAL)
+            self.update, SCAN_INTERVAL
         )
 
         done = await self._async_attach_player()
```

Comment:
In `_async_attach_player`
```
            await self._subscribe(player.deviceProperties, self.async_update_properties)
```

Then consume the dispatcher this in the sensor (we can move it later to __init__ in a future PR)

```

    @callback
    def async_update_properties(self, event: SonosEvent | None = None) -> None:
        """Update information from properties."""
        if not event:
            return
        async_dispatcher_send(self.hass, f"{SONOS_PROPERTIES_UPDATE}-{self._player.uid}", event)
```